### PR TITLE
GWIS configuration

### PR DESCRIPTION
**Pull request by joshua-slaughter**
Hi all, I have put together a configuration for estimating pairwise interactions at the genome-wide scale. The changes to make this possible only required minimal changes to `inputs_from_config.jl`.  A review would be very appreciated!

### CODE REVIEW

File: test/inputs_from_gwis_config.jl
Comment (olivierlabayle) [inline test/inputs_from_gwis_config.jl:1]: This test file has no obvious difference with the GWAS mode, however I would expect the generated estimands to be quite different. Also, you can reuse a test function like `get_summary_stats` instead of redefining it it is exactly the same by including it in the `testutils.jl` file.

File: src/inputs_from_config.jl [hunk 1]
```diff
@@ -60,13 +60,22 @@ function estimands_from_variants(
...
     estimands = []
     for variant in variants
-        treatments = treatments_from_variant(variant, dataset)
+       
+        if isempty(extra_treatments)
+            treatments = treatments_from_variant(variant, dataset)
+        elseif length(extra_treatments) == 1
+            treatments = Dict(treatments_from_variant(variant, dataset)..., treatments_from_variant(string(extra_treatments[1]), dataset)...)
+        else
+            error("GWIS mode only supports pairwise interaction with one extra treatment.")
```
Comment (olivierlabayle) [inline src/inputs_from_config.jl:76]: if someone tries to do a GWAS with extra treatment this error message will be misleading.

**Review by olivierlabayle — CHANGES_REQUESTED** [review]
Thanks Josh, see my comments in the review, happy to discuss.

File: src/inputs_from_config.jl [hunk 2]
```diff
@@ -60,13 +60,22 @@ function estimands_from_variants(
...
     outcome_extra_covariates=[],
     positivity_constraint=0.,
     verbosity=1
     )
     estimands = []
     for variant in variants
-        treatments = treatments_from_variant(variant, dataset)
+       
+        if isempty(extra_treatments)
+            treatments = treatments_from_variant(variant, dataset)
```
[the remark this answers is not in the document]
Comment (joshua-slaughter) [reply src/inputs_from_config.jl:72]: Circling back to this! Sorry for the delay. The goal of this mode is to allow for the testing of interactions with specified set of environmental variables genome-wide. The current GWAS functionality already supports testing all variants in the bed files without need to explicitly state them as `extra_treatments`. Because of think an appropriate naming would be `genome_wide_environmental_interaction_study`?

File: src/inputs_from_config.jl [hunk 3]
```diff
@@ -60,13 +60,22 @@ function estimands_from_variants(
...
+        elseif length(extra_treatments) == 1
```
[the remark this answers is not in the document]
Comment (joshua-slaughter) [reply src/inputs_from_config.jl:73]: Will do!

File: src/inputs_from_config.jl (hunk 1 above)
Comment (joshua-slaughter) [reply src/inputs_from_config.jl:76]: This will be resolved in the generalization of the function to handle multiple treatments.

File: src/inputs_from_config.jl [hunk 4]
```diff
@@ -187,6 +198,24 @@ function estimands_from_gwas(dataset, variants, outcomes, confounders;
     return vcat(estimands_partitions...)
 end
 
+function estimands_from_gwis(dataset, variants, outcomes, confounders;
```
[the remark this answers is not in the document]
Comment (joshua-slaughter) [reply src/inputs_from_config.jl:201]: Agreed

**Comment by olivierlabayle** [comment]
@joshua-slaughter Could you provide a high level overview of the new functionality here and how a user is meant to interact with it? This description could also be used in the next release of the TarGene docs.

File: test/inputs_from_gwis_config.jl [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+    for Ψ in estimands
+        # Get the treatment keys from the first arg
+        treatment_set = Set(keys(Ψ.args[1].treatment_values))
+        
+        # Find the variant (treatment that's not in extra_treatments)
+        variants = setdiff(treatment_set, extra_treatments)
+        
+        # Skip if no variant found (shouldn't happen, but defensive)
+        isempty(variants) && continue
```
Comment (olivierlabayle) [inline test/inputs_from_gwis_config.jl]: No this should be a test, e.g., tests are your defense against bug so you need to catch them:

```
@test length(variants) == 1
```

**Review by olivierlabayle — CHANGES_REQUESTED** [review]
Hi Josh, thanks for the PR, please find my comments inline. Main things are:

1. I think the summary stats (counts) should either be moved to TMLECLI as an option or kept here but in a different function. It would also be good to discuss exactly what we want to countI suggest to make use of DataFrames.jl groupby combine in any case to simplify the code and make it faster.

2. The GWIS looks mostly good while I think it is getting quite complicated and requires a bit more testing of edge cases that are not covered.

Thanks!

File: test/data/config_gwis_first_order.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+type: gwas
```
[the remark this answers is not in the document]
Comment (joshua-slaughter) [reply test/data/config_gwis_first_order.yaml]: Sorry yeah I extended it to handle many cases. Will rename.

File: src/inputs_from_config.jl [hunk 7]
```diff
@@ -207,20 +255,97 @@ function read_bed_chromosome(bedprefix)
...
-    genotypes = DataFrame(convert(Matrix{UInt8}, snpdata.snparray), snpdata.snp_info."snpid")
+    genotypes = DataFrame(convert(Matrix{UInt8}, snpdata.snparray), snpdata.snp_info."snpid"; makeunique=true)
+    genotype_ids = names(genotypes)
     genotype_map = Union{UInt8, Missing}[0, missing, 1, 2]
-    for col in names(genotypes)
+    for col in genotype_ids
         genotypes[!, col] = [genotype_map[x+1] for x in genotypes[!, col]]
     end
     insertcols!(genotypes, 1, :SAMPLE_ID => snpdata.person_info."iid")
+
```
[the remark this answers is not in the document]
Comment (joshua-slaughter) [reply src/inputs_from_config.jl]: Okay I'll take note of this and make a new PR for the CLI. Agreed on the efficiency part as well. Also, this function is only really used for GWAS as well see `make_genotypes`
